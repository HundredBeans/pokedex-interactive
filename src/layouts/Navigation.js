import React from "react";
import {
  Flex,
  Box,
  Heading,
  Link,
  useColorModeValue,
  useBreakpointValue,
  Icon,
} from "@chakra-ui/react";
import { motion } from "framer-motion";
import { Link as RouterLink } from "react-router-dom";
import ColorModeSwitcher from "../components/ColorModeSwitcher";
import HeaderLogo from "../components/HeaderLogo";
import { CgPokemon, CgSearch, CgList } from "react-icons/cg";

function Navigation(props) {
  const navColor = useColorModeValue("#FAF5FF", "#1A202C");
  const MotionHeading = motion(Heading);
  const isMd = useBreakpointValue({ base: true, md: false });

  return (
    <React.Fragment>
      {/* Top Navigation */}
      <header>
        <Flex
          justifyContent="space-between"
          p="4"
          position="fixed"
          height={props.headerHeight}
          width="100vw"
          shadow="lg"
          zIndex="banner"
          background={navColor}
        >
          <HeaderLogo />
          <Flex align="center" justify="center" justifyContent="space-between">
            {!isMd ? (
              <React.Fragment>
                <Link as={RouterLink} to="/listofall" _hover="none">
                  <MotionHeading
                    whileHover={{ opacity: 0.5 }}
                    p={5}
                    fontSize="lg"
                  >
                    List of All Pokemons
                  </MotionHeading>
                </Link>
                <Link as={RouterLink} to="/pokedetails" _hover="none">
                  <MotionHeading
                    whileHover={{ opacity: 0.5 }}
                    p={5}
                    fontSize="lg"
                  >
                    Battle
                  </MotionHeading>
                </Link>
                <Link as={RouterLink} to="/about" _hover="none">
                  <MotionHeading
                    whileHover={{ opacity: 0.5 }}
                    p={5}
                    fontSize="lg"
                  >
                    About
                  </MotionHeading>
                </Link>
              </React.Fragment>
            ) : (
              ""
            )}
            <Box>
              <ColorModeSwitcher></ColorModeSwitcher>
            </Box>
          </Flex>
        </Flex>
      </header>
      {/* Bottom Navigation */}
      <Box
        bottom="0"
        position="fixed"
        width="100%"
        background={navColor}
        zIndex="banner"
        shadow="lg"
        height="70px"
        py={3}
        display={!isMd ? "none" : ""}
      >
        <Flex justifyContent="space-around">
          <Link as={RouterLink} _active={{ color: "green" }} to="/example">
            <Icon fontSize="5xl" as={CgList} />
          </Link>
          <Link as={RouterLink} _activeLink={{ opacity: 0.5 }} to="/">
            <Icon fontSize="5xl" as={CgPokemon} />
          </Link>
          <Link as={RouterLink} _activeLink={{ opacity: 0.5 }} to="/">
            <Icon fontSize="5xl" as={CgSearch} />
          </Link>
        </Flex>
      </Box>
    </React.Fragment>
  );
}

export default Navigation;
