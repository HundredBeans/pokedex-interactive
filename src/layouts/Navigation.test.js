import React from "react";
import Navigation from "./Navigation";
import { render } from "../test-utils";
import { cleanup } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { waitFor } from "@testing-library/dom";

// Mock history function https://stackoverflow.com/questions/58524183/how-to-mock-history-push-with-the-new-react-router-hooks-using-jest
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    location: {
      pathname: "pathname/mocked",
    },
  }),
  matchPath: () => ({
    params: {
      id: "mock-id",
    },
  }),
}));

describe("Navigation.js test", () => {
  afterEach(() => {
    cleanup();
  });

  it("renders the component", async () => {
    const renderedComponent = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );
    await waitFor(() => {
      expect(renderedComponent).toMatchSnapshot();
    });
  });
});
