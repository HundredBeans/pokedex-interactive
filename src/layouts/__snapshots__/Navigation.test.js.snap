// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation.js test renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; top: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; top: 0px; right: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto; bottom: 0px; right: 0px; left: 0px;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; bottom: 0px; right: 0px;"
      />
    </div>
    <div>
      <header>
        <div
          class="css-1mgnoqi"
        >
          <div
            class="css-gmuwbf"
          >
            <div
              class="css-gmuwbf"
            >
              <button
                class="chakra-button css-rt2cw"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-xj7snx"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
                  />
                </svg>
              </button>
            </div>
            <h2
              class="chakra-heading css-1y42k7"
            >
              MOCK-ID
            </h2>
          </div>
          <div
            class="css-1lekzkb"
          >
            <a
              class="chakra-link css-1qlgi3u"
              href="/pokemons"
            >
              <h2
                class="chakra-heading css-mdwrt3"
              >
                Pokemons List
              </h2>
            </a>
            <a
              class="chakra-link css-1qlgi3u"
              href="/my-pokemons"
            >
              <h2
                class="chakra-heading css-mdwrt3"
              >
                My Pokemons
              </h2>
            </a>
            <div
              class="css-0"
            >
              <button
                aria-label="Switch to dark mode"
                class="chakra-button css-18ehdyv"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="css-b7tyx2"
      >
        <div
          class="css-1bnxa12"
        >
          <a
            class="chakra-link css-1990i9s"
            href="/pokemons"
          >
            <svg
              class="chakra-icon css-xj7snx"
              fill="none"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="8"
                x2="21"
                y1="6"
                y2="6"
              />
              <line
                x1="8"
                x2="21"
                y1="12"
                y2="12"
              />
              <line
                x1="8"
                x2="21"
                y1="18"
                y2="18"
              />
              <line
                x1="3"
                x2="3"
                y1="6"
                y2="6"
              />
              <line
                x1="3"
                x2="3"
                y1="12"
                y2="12"
              />
              <line
                x1="3"
                x2="3"
                y1="18"
                y2="18"
              />
            </svg>
            <br />
            <span>
              Pokemon List
            </span>
          </a>
          <a
            class="chakra-link css-1990i9s"
            href="/my-pokemons"
          >
            <svg
              class="chakra-icon css-xj7snx"
              fill="none"
              focusable="true"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
              <path
                d="M23 21v-2a4 4 0 0 0-3-3.87"
              />
              <path
                d="M16 3.13a4 4 0 0 1 0 7.75"
              />
            </svg>
            <br />
            <span>
              My Pokemons
            </span>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="css-1mgnoqi"
      >
        <div
          class="css-gmuwbf"
        >
          <div
            class="css-gmuwbf"
          >
            <button
              class="chakra-button css-rt2cw"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-xj7snx"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
                />
              </svg>
            </button>
          </div>
          <h2
            class="chakra-heading css-1y42k7"
          >
            MOCK-ID
          </h2>
        </div>
        <div
          class="css-1lekzkb"
        >
          <a
            class="chakra-link css-1qlgi3u"
            href="/pokemons"
          >
            <h2
              class="chakra-heading css-mdwrt3"
            >
              Pokemons List
            </h2>
          </a>
          <a
            class="chakra-link css-1qlgi3u"
            href="/my-pokemons"
          >
            <h2
              class="chakra-heading css-mdwrt3"
            >
              My Pokemons
            </h2>
          </a>
          <div
            class="css-0"
          >
            <button
              aria-label="Switch to dark mode"
              class="chakra-button css-18ehdyv"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="css-b7tyx2"
    >
      <div
        class="css-1bnxa12"
      >
        <a
          class="chakra-link css-1990i9s"
          href="/pokemons"
        >
          <svg
            class="chakra-icon css-xj7snx"
            fill="none"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="8"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="8"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="8"
              x2="21"
              y1="18"
              y2="18"
            />
            <line
              x1="3"
              x2="3"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="3"
              y1="18"
              y2="18"
            />
          </svg>
          <br />
          <span>
            Pokemon List
          </span>
        </a>
        <a
          class="chakra-link css-1990i9s"
          href="/my-pokemons"
        >
          <svg
            class="chakra-icon css-xj7snx"
            fill="none"
            focusable="true"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M23 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
          <br />
          <span>
            My Pokemons
          </span>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
